### PR TITLE
feat(api-markdown-documenter): Add support for multiple package entry-points

### PR DESCRIPTION
**Pull request by Josmithr**
For packages with a single entry-point, the generated output structure remains unchanged. For packages with multiple entry points, an extra document will be generated for each entry-point under the package directory.
- Note that we still will only generate at most 1 document for each API item, regardless of how many entry-points in which they appear.

Includes a unit test for the scenario.

Note: API-Extractor doesn't currently support this, but it is planned, and the API report format supports it.

### Breaking Change

These changes constitute a breaking change due to changing transformation config options (removes `transformApiPackage` and adds `transformApiEntryPoint`).

### CODE REVIEW

File: tools/api-markdown-documenter/src/test/MarkdownDocumenter.test.ts [hunk 1]
```diff
@@ -143,18 +142,6 @@ function apiTestSuite(
 					);
 				});
 
-				it("Render Model document (smoke test)", () => {
```
Comment (Josmithr) [inline tools/api-markdown-documenter/src/test/MarkdownDocumenter.test.ts:146]: These tests were useful early on before we had better test coverage, but they no longer serve much purpose, and would require exporting things that shouldn't be exported.

File: tools/api-markdown-documenter/src/api-item-transforms/TransformApiPackage.ts [hunk 2]
```diff
@@ -1,46 +0,0 @@
-/*!
```
Comment (Josmithr) [inline tools/api-markdown-documenter/src/api-item-transforms/TransformApiPackage.ts:1]: This logic moved into `api-item-transforms/TransformApiModel.ts`

File: tools/api-markdown-documenter/src/api-item-transforms/TransformModel.ts [hunk 3]
```diff
@@ -1,35 +0,0 @@
...
-
-/**
- * Generates a {@link DocumentNode} for the specified `apiModel`.
- *
- * @param apiModel - The API model content to be rendered. Represents the root of the API suite.
- * @param config - See {@link MarkdownDocumenterConfiguration}.
- *
- * @returns The rendered Markdown document.
- */
-export function apiModelToDocument(
```
Comment (Josmithr) [inline tools/api-markdown-documenter/src/api-item-transforms/TransformModel.ts:19]: This logic moved into `api-item-transforms/TransformApiModel.ts`

File: tools/api-markdown-documenter/src/api-item-transforms/TransformApiPackage.ts [hunk 4]
```diff
@@ -1,46 +0,0 @@
...
-
-/**
- * Creates a {@link DocumentNode} for the specified `apiPackage`.
- *
- * @param apiPackage - The package content to be rendered.
- * @param config - See {@link MarkdownDocumenterConfiguration}.
- *
- * @returns The rendered Markdown document.
- */
-export function apiPackageToDocument(
```
Comment (Josmithr) [inline tools/api-markdown-documenter/src/api-item-transforms/TransformApiPackage.ts:21]: This logic moved into `api-item-transforms/TransformApiModel.ts`, and was made a private implementation detail.

File: tools/api-markdown-documenter/src/MarkdownDocumenter.ts [hunk 5]
```diff
@@ -1,195 +0,0 @@
...
- *
- * The generated nodes' {@link DocumentNode.filePath}s are determined by the provided output path and the
- * following configuration properties:
- *
- * - {@link DocumentationSuiteOptions.documentBoundaries}
- * - {@link DocumentationSuiteOptions.hierarchyBoundaries}
- *
- * @param transformConfig - Configuration for transforming API items into {@link DocumentationNode}s.
- */
-export function transformApiModel(
```
Comment (Josmithr) [inline tools/api-markdown-documenter/src/MarkdownDocumenter.ts:54]: This logic moved into `api-item-transforms/TransformApiModel.ts`.

File: tools/api-markdown-documenter/src/MarkdownDocumenter.ts [hunk 6]
```diff
@@ -1,195 +0,0 @@
...
- * following configuration properties:
- *
- * - {@link DocumentationSuiteOptions.documentBoundaries}
- * - {@link DocumentationSuiteOptions.hierarchyBoundaries}
- *
- * @param transformConfig - Configuration for transforming API items into {@link DocumentationNode}s.
- * @param renderConfig - Configuration for rendering {@link DocumentNode}s as Markdown.
- * @param outputDirectoryPath - The directory under which the document files will be generated.
- */
-export async function renderApiModelAsMarkdown(
```
Comment (Josmithr) [inline tools/api-markdown-documenter/src/MarkdownDocumenter.ts:130]: This and `renderDocumentsAsMarkdown` moved to `RenderMarkdown.ts`.

File: tools/api-markdown-documenter/src/api-item-transforms/TransformApiModel.ts [hunk 7]
```diff
@@ -0,0 +1,270 @@
...
+	// Render body contents
+	const sections = createModelBodySections(apiModel, config);
+
+	logger.verbose(`API Model document rendered successfully.`);
+
+	return createDocument(apiModel, sections, config);
+}
+
+/**
+ * TODO
```
Comment (tylerbutler) [inline tools/api-markdown-documenter/src/api-item-transforms/TransformApiModel.ts:169]: For this PR?

File: tools/api-markdown-documenter/src/api-item-transforms/TransformApiModel.ts [hunk 8]
```diff
@@ -0,0 +1,270 @@
...
+ *
+ * @param apiPackage - The package content to be rendered.
+ * @param apiEntryPoint - The package content to be rendered.
```
Comment (tylerbutler) [inline tools/api-markdown-documenter/src/api-item-transforms/TransformApiModel.ts:172]: Are the docs meant to be the same?

File: tools/api-markdown-documenter/src/api-item-transforms/TransformApiModel.ts (hunk 7 above)
Comment (tylerbutler) [reply tools/api-markdown-documenter/src/api-item-transforms/TransformApiModel.ts:169]: there are some other TODOs elsewhere, same question.

**Review by tylerbutler — APPROVED** [review]
Very minor nits.

File: tools/api-markdown-documenter/src/api-item-transforms/TransformApiModel.ts (hunk 8 above)
Comment (Josmithr) [reply tools/api-markdown-documenter/src/api-item-transforms/TransformApiModel.ts:172]: Oof, no they're not. Will clean up.

File: tools/api-markdown-documenter/src/api-item-transforms/TransformApiModel.ts (hunk 7 above)
Comment (Josmithr) [reply tools/api-markdown-documenter/src/api-item-transforms/TransformApiModel.ts:169]: Ah, yup. Need to fill those in.

**Pull request merged**